src/main/java/net/openhft/chronicle/queue/impl/single/Pretoucher.java

```diff
@@ -126,27 +126,12 @@ private void assignCurrentCycle() {
         if (qCycle != currentCycle) {
             releaseResources();
 
-            if (canWrite)
-                queue.writeLock().lock();
-            try {
-                if (!earlyAcquireNextCycle && currentCycleWireStore != null && canWrite)
-                    try {
-                        final Wire wire = queue.wireType().apply(currentCycleMappedBytes);
-                        wire.usePadding(currentCycleWireStore.dataVersion() > 0);
-                        currentCycleWireStore.writeEOF(wire, queue.timeoutMS);
-                    } catch (Exception ex) {
-                        Jvm.warn().on(getClass(), "unable to write the EOF file=" + currentCycleMappedBytes.mappedFile().file(), ex);
-                    }
-                SingleChronicleQueueStore oldStore = currentCycleWireStore;
-                currentCycleWireStore = queue.storeForCycle(qCycle, queue.epoch(), earlyAcquireNextCycle || canWrite, currentCycleWireStore);
-                if (oldStore != null && oldStore != currentCycleWireStore)
-                    oldStore.close();
-            } finally {
-                if (canWrite)
-                    queue.writeLock().unlock();
-            }
+            SingleChronicleQueueStore oldStore = currentCycleWireStore;
```

Comment:
After `releaseResources()`, `currentCycleWireStore` will be null, won't it?

Reply:
good point

Reply:
fixed